cavern/src/main/java/org/opencadc/cavern/files/FileAction.java

```diff
@@ -162,14 +162,25 @@ public void initAction() throws Exception {
 
         // Initialize the nodeURI value. Check authorization, either token validation or
         // user authentication against node attributes.
-        String path = syncInput.getPath();
-        if (isPreauth == true) {
-            initPreauthTarget(path);
-        } else {
-            initAuthTarget(path);
+        // Exceptions from init functions are put to syncOutput so they can
+        // be reported to the caller.
+        try {
+            String path = syncInput.getPath();
+            if (isPreauth == true) {
+                initPreauthTarget(path);
+            } else {
+                initAuthTarget(path);
+            }
+        } catch (NodeNotFoundException e) {
```

Comment:
It is correct that you are catching these vospace specific exceptions, but instead of doing the error handling manually, you can just rethrow the mapped exception and let RestAction.java handle them.  So there you would throw a new ResourceNotFoundException(e), and in the next block a new IllegalArgumentException(e).  Shouldn't need the last catch for IllegalArgumentException.

Reply:
If that works you can get rid of sendError()

Reply:
think this is all done now.